v4.11.0 Plugin Crash

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

```
Terraform v1.8.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v5.72.0
+ provider registry.terraform.io/okta/okta v4.11.0

Your version of Terraform is out of date! The latest version
is 1.9.7. You can update by downloading from https://www.terraform.io/downloads.html
```

### Affected Resource(s)
- okta_group_owner

### Terraform Configuration Files
n/a

### Debug Output

<details>
  <summary>Click to expand</summary>
  
```
Stack trace from the terraform-provider-okta_v4.11.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x1056911d8]

goroutine 1182 [running]:
github.com/okta/terraform-provider-okta/okta.mapGroupOwnerToState(0x140002a5908?, 0x105f2d4f8?)
	github.com/okta/terraform-provider-okta/okta/resource_okta_group_owner.go:228 +0x18
github.com/okta/terraform-provider-okta/okta.(*groupOwnerResource).Read(0x140001760b0, {0x105f2d4f8, 0x14000c03b90}, {{{{0x105f32a60, 0x14001087ad0}, {0x105bd63c0, 0x14001087950}}, {0x105f356b0, 0x14000734050}}, 0x140001760c0, ...}, ...)
	github.com/okta/terraform-provider-okta/okta/resource_okta_group_owner.go:150 +0x3dc
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0x14000275520, {0x105f2d4f8, 0x14000c03b90}, 0x14000c03c20, 0x140006e9560)
	github.com/hashicorp/terraform-plugin-framework@v1.8.0/internal/fwserver/server_readresource.go:101 +0x514
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0x14000275520, {0x105f2d4f8?, 0x14000c03a10?}, 0x14000b29780)
	github.com/hashicorp/terraform-plugin-framework@v1.8.0/internal/proto6server/server_readresource.go:55 +0x314
github.com/hashicorp/terraform-plugin-mux/tf6to5server.v6tov5Server.ReadResource({{0x105f36098?, 0x14000275520?}}, {0x105f2d4f8?, 0x14000c03a10?}, 0x14000b295c0?)
	github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf6to5server/tf6to5server.go:238 +0x228
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadResource(0x105f2d530?, {0x105f2d4f8?, 0x14000c03710?}, 0x14000b295c0)
	github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5muxserver/mux_server_ReadResource.go:35 +0x184
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x140002a6e60, {0x105f2d4f8?, 0x14000ac4960?}, 0x14000eb25a0)
	github.com/hashicorp/terraform-plugin-go@v0.22.2/tfprotov5/tf5server/server.go:775 +0x3c4
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x105e5f2c0?, 0x140002a6e60}, {0x105f2d4f8, 0x14000ac4960}, 0x14000bb8700, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.22.2/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:482 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001b3200, {0x105f2d4f8, 0x14000ac48d0}, {0x105f33cd8, 0x140001c4180}, 0x14000b8cea0, 0x14000492030, 0x106dfe908, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xba0
google.golang.org/grpc.(*Server).handleStream(0x140001b3200, {0x105f33cd8, 0x140001c4180}, 0x14000b8cea0)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0xc80
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 15
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x150

Error: The terraform-provider-okta_v4.11.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2024-10-16T15:56:27.674+1100 [DEBUG] provider: plugin exited
```

</details>

### Panic Output

### Expected Behavior
No Crash

#### Can this be done in the Admin UI?
N/A

#### Can this be done in the actual API call?
N/A

### Actual Behavior
crash on terraform plan

### Steps to Reproduce
Unclear at this time, provider crashes on terraform plan in our staging env

### Important Factoids

### References
-

--- a/okta/config.go
+++ b/okta/config.go
@@ -26,7 +26,7 @@ import (
 )
 
 const (
-	OktaTerraformProviderVersion   = "4.11.0"
+	OktaTerraformProviderVersion   = "4.11.1"
 	OktaTerraformProviderUserAgent = "okta-terraform/" + OktaTerraformProviderVersion
 )
 

--- a/okta/resource_okta_group_owner.go
+++ b/okta/resource_okta_group_owner.go
@@ -132,7 +132,6 @@ func (r *groupOwnerResource) Read(ctx context.Context, req resource.ReadRequest,
 	var err error
 
 	listGroupOwners, _, err := r.Config.oktaSDKClientV5.GroupOwnerAPI.ListGroupOwners(ctx, state.GroupID.ValueString()).Execute()
-
 	if err != nil {
 		resp.Diagnostics.AddError(
 			"Error retrieving list group owners",
@@ -144,7 +143,14 @@ func (r *groupOwnerResource) Read(ctx context.Context, req resource.ReadRequest,
 	for _, groupOwner := range listGroupOwners {
 		if groupOwner.GetId() == state.ID.ValueString() {
 			grpOwner = &groupOwner
+			break
 		}
+	}
+
+	if grpOwner == nil {
+		// The resource no longer exists; remove it from state
+		resp.State.RemoveResource(ctx)
+		return
 	}
 
 	resp.Diagnostics.Append(mapGroupOwnerToState(grpOwner, &state)...)

